src/hotspot/share/jvmci/jvmciCompilerToVM.cpp

```diff
@@ -70,6 +70,11 @@
 #include "jfr/jfr.hpp"
 #endif
 
+#if defined(TARGET_COMPILER_gcc)
+#undef JNIEXPORT
```

Comment:
Based on the suggestion in [this JBS comment](https://bugs.openjdk.org/browse/JDK-8017234?focusedId=14601421&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14601421), I think we can remove JNIEXPORT from all the functions in this file.

We need to keep the JNICALL though; the deprecated Win32 port needs it, not sure if it's used anywhere else.

Reply:
I think so too. 

However, I would *really* like to do that separately as a follow up. My goal with this PR has been to make no changes in the product. Consider what would happen if we removed this, and there was some strange failure down the line. There is a high risk the entire PR would be backed out. 

If instead we remove the JNIEXPORT in a separate change, and this causes failures, then it is a much easier backout, and we would be much better informed at exactly what failed.